Testnet/StandardTokenOwnableDynamicSupply/StandardTokenOwnableDynamicSupply/StandardTokenOwnableDynamicSupplyContract.cs

```diff
@@ -0,0 +1,265 @@
+﻿using Stratis.SmartContracts;
+using Stratis.SmartContracts.Standards;
+
+/// <summary>
+/// Implementation of a standard token contract for the Stratis Platform.
+/// </summary>
+[Deploy]
+public class StandardTokenOwnableDynamicSupplyContract : SmartContract, IStandardToken256, IOwnable, IMintable, IBurnable, IBurnableWithMetadata
+{
+    /// <summary>
+    /// Constructor used to create a new instance of the token. Assigns the total token supply to the creator of the contract.
+    /// </summary>
+    /// <param name="smartContractState">The execution state for the contract.</param>
+    /// <param name="totalSupply">The total token supply.</param>
+    /// <param name="name">The name of the token.</param>
+    /// <param name="symbol">The symbol used to identify the token.</param>
+    /// <param name="decimals">The amount of decimals for display and calculation purposes.</param>
+    public StandardTokenOwnableDynamicSupplyContract(ISmartContractState smartContractState, UInt256 totalSupply, string name, string symbol, byte decimals) : base(smartContractState)
+    {
+        this.TotalSupply = totalSupply;
+        this.Name = name;
+        this.Symbol = symbol;
+        this.Decimals = decimals;
+        this.SetBalance(Message.Sender, totalSupply);
+        this.Owner = Message.Sender;
```

Comment:
This specifically can be problematic with the `MultisigContract` - https://github.com/stratisproject/CirrusSmartContracts/pull/59/files

The Multisig contract can't create this token contract as its currently written. Maybe adding an `owner` param to this constructor and remove the assignment using `Message.Sender`

Reply:
Just so that I'm not misunderstanding, why would it not be able to create it? Wouldn't the multisig contract address be `Message.Sender`? Or is the behaviour of sender different during creation from within another contract?

Presumably if the multisig contract were to create instances of this one directly then there would need to be a dedicated method added to the multisig for it, as the transaction submission mechanism can't handle contract creation. And eventually a method to deploy the NFT contract would need to be added as well.

Reply:
The MS contract/project will need to include all code from this contract if is to be able to create these enhanced tokens and would need a method internally to call the `Create` method. Then the MS contract would properly be the owner of this contract which enables it to `Mint` tokens. 

As these are two separate contracts currently, the MS contract cannot create any contracts, so it can't be the owner of an enhanced token, so it can't mint new tokens for the supply. 

Same for NFT's then I suppose if they're assigning owners right at creation based on Message.Sender.

Reply:
Well, the way this was handled for wSTRAX is that the ERC20 was deployed by a normal EOA and then transferred to the multisig as a second transaction. This approach works on Cirrus too.

The ability for the multisig to collectively agree to create contracts by vote would be a nice to have, but as you say the project would then need every contract it would ever need to deploy inside it, and could never be extended to support additional contract types.

Reply:
ahh I see, so you're saying this contract could be created then would use `TransferOwnership` to the multisig contract? That should be fine, wasn't considering that route.